test/assert_test_m.F90

```diff
@@ -0,0 +1,100 @@
+! Copyright (c) 2024, The Regents of the University of California and Sourcery Institute
+! Terms of use are as specified in LICENSE.txt
+
+#include "language-support.F90"
+#include "julienne-assert-macros.h"
+
+module assert_test_m
+  !! Test Julienne's assert generic interface
+
+  use julienne_m, only : & 
+     assertion_diagnosis_t &
+    ,call_julienne_assert_ &
+    ,test_diagnosis_t &
+    ,test_t &
+    ,test_description_t &
+    ,test_description_substring &
+    ,test_result_t &
+    ,operator(.approximates.) &
+    ,operator(.within.)
+  implicit none
+
+  private
+  public :: assert_test_t
+
+  type, extends(test_t) :: assert_test_t
+  contains
+    procedure, nopass :: subject
+    procedure, nopass :: results
+  end type
+
+contains
+
+  pure function subject() result(specimen)
+    character(len=:), allocatable :: specimen
+    specimen = "The julienne_assert subroutine" 
+  end function
+
+#if HAVE_PROCEDURE_ACTUAL_FOR_POINTER_DUMMY
+
+  function results() result(test_results)
+    type(test_result_t), allocatable :: test_results(:)
+
+    associate(descriptions => [ &
+       test_description_t("invocation via macro with a test_diagnosis_t expression", check_macro_with_expression) &
+      ,test_description_t("invocation via macro with an assertion_diagnosis_t argument", check_macro_with_assertion_diagnosis) &
+    ])
+      associate(substring_in_subject => index(subject(), test_description_substring) /= 0)
+        associate(substring_in_assertion_diagnosis => descriptions%contains_text(test_description_substring))
+          associate(matching_descriptions => pack(descriptions, substring_in_subject .or. substring_in_assertion_diagnosis))
```

Comment:
This is probably orthogonal to this PR: 

IIUC this code (and the much longer version below for gfortran) is Julienne's recommended boilerplate for supporting `--contains` test filtering, which IS a nice feature... but shoving this command-line implementation goop into every test seems ugly and potentially unnecessary.

Doesn't Julienne control the `run()` method invoked below? Why can't Julienne centralize and hide this filtering logic inside that `run()` method and just no-op the call away when the test is filtered out?